src/js/common/schemaform/validation.js

```diff
@@ -151,9 +151,9 @@ export function uiSchemaValidate(errors, uiSchema, schema, formData, path = '')
       validations.forEach(validation => {
         const pathErrors = path ? _.get(path, errors) : errors;
         if (typeof validation === 'function') {
-          validation(pathErrors, currentData, formData, schema, uiSchema['ui:errorMessages']);
+          validation(pathErrors, currentData, formData, schema, uiSchema['ui:errorMessages'], currentIndex);
```

Comment:
It's so we can conditionally validate data in an array. Previously we hadn't had a situation where some data in an array item needed to be validated in combination with other data in that item (this happens for child dependent date).